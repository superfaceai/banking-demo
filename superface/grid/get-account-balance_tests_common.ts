import { SuperfaceClient } from "@superfaceai/one-sdk";

export function getAccountBalanceTest(providerName: string) {
    describe(`banking/get-account-balance/${providerName}`, () => {
        it('returns a result when called with ...', async () => {
            const sdk = new SuperfaceClient();
            const profile = await sdk.getProfile('banking/get-account-balance');
            const provider = await sdk.getProvider(providerName);

            const result = await profile
                .getUseCase('GetAccountBalance')
                .perform(undefined, { provider });

            expect(result.isOk()).toBe(true);
            expect(result.unwrap()).toMatchSnapshot();
        });
    });
}