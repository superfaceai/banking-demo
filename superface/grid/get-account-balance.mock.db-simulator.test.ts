import { getAccountBalanceTest } from "./get-account-balance_tests_common";

getAccountBalanceTest('db-simulator');