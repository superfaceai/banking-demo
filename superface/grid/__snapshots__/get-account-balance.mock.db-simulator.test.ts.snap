// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banking/get-account-balance/db-simulator returns a result when called with ... 1`] = `
Object {
  "accounts": Array [
    Object {
      "accountType": "CURRENT_ACCOUNT",
      "bic": "DEUTDEFFXXX",
      "currencyCode": "EUR",
      "currentBalance": 100.95,
      "iban": "DE10010000000000009318",
      "productDescription": "Pers√∂nliches Konto",
    },
  ],
}
`;
